aten/src/ATen/native/Distributions.h

```diff
@@ -17,6 +17,15 @@ namespace {
 #define isnan std::isnan
 #endif
 
+// workaround: cuda 8 cannot do static_cast with half
+#if (defined(__CUDACC_VER_MAJOR__) && (__CUDACC_MAJOR_VER < 9))
+template<typename R, typename T>
+deviceforcuda R cast_wrapper(T v) { return scalar_cast<R>(v); }
+#else
```

Comment:
scalar_cast works both for cuda 8 and cuda 9, so you can just use it everywhere without wrapping.

Reply:
But it doesn't non-cuda, does it?

Reply:
So that I change the ifdef to `#ifdef __CUDACC__ ` to have scalar_cast for all of them?
I must admit I thought that the static_cast was the more standard way.